src/services/ExperianService.ts

```diff
@@ -101,40 +133,51 @@ export class ExperianService {
     	const storedToken = await this.getExperianToken();
     	const isValidToken = this.checkExperianToken(storedToken);
 
-    	if (isValidToken) {
+    	if (storedToken && isValidToken) {
+    		this.logger.info("Valid token found");
     		return storedToken;
-    	} else {		
-    		this.logger.info("requestExperianToken - no valid token - trying to generate new Experian token");
+
+    	} else {	
     		try {
+    			const endpoint = `${this.experianBaseUrl}${Constants.EXPERIAN_TOKEN_ENDPOINT_PATH}`;
+    			this.logger.info("No valid token found - trying to generate new Experian token", { endpoint });
     			const params = {
     				username: clientUsername,
     				password: clientPassword,
     				client_id: clientId,
     				client_secret: clientSecret,
     			};
 
+    			const correlationId = randomUUID();
     			const config: AxiosRequestConfig<any> = {
     				headers: {
     					"Content-Type": "application/json",
-    					"X-Correlation-Id": randomUUID(),
+    					"X-Correlation-Id": correlationId,
     					"X-User-Domain": "cabinetofficegds.com",
     				},
     			};
 
     			const { data }: { data: ExperianTokenResponse } = await axios.post(
-    				`${this.experianBaseUrl}${Constants.EXPERIAN_TOKEN_ENDPOINT_PATH}`,
+    				endpoint,
     				params,
     				config,
     			);
-    			this.logger.info("Received response from experian token endpoint");
+    			this.logger.info(`Received response from Experian token endpoint - X-Correlation-Id: ${correlationId}`);
     			await this.saveExperianToken(data);
     			return data;
+
     		} catch (error: any) {
-    			this.logger.error({ message: "Error generating experian token", statusCode: error?.response?.status, messageCode: MessageCodes.FAILED_GENERATING_EXPERIAN_TOKEN });
-    			this.logger.info("Returning previous Experian token due to error generating a new one.");
-    			return storedToken;
+    			if (storedToken) {
+    				const message = "Error generating Experian token - returning previous Experian token";
```

Comment:
```suggestion
    				const message = "Error refreshing Experian token - returning previous Experian token";
```
Being really clear about what action you've done in the error messages will save you headaches later on when the code isnt fresh and there's a issue